sklearn/model_selection/tests/test_search.py

```diff
@@ -332,6 +333,36 @@ def test_grid_search_groups():
         gs.fit(X, y)
 
 
+def test_return_train_score_warn():
+    # Test that warnings are raised. Will be removed in 0.21
+
+    X = np.arange(100).reshape(10, 10)
+    y = np.array([0] * 5 + [1] * 5)
+    grid = {'C': [1, 2]}
+
+    estimators = [GridSearchCV(LinearSVC(random_state=0), grid),
+                  RandomizedSearchCV(LinearSVC(random_state=0), grid,
+                                     n_iter=2)]
+
+    result = {}
+    for estimator in estimators:
+        for val in [True, False, 'warn']:
+            estimator.set_params(return_train_score=val)
+            result[val] = assert_no_warnings(estimator.fit, X, y).cv_results_
+
+    for key in ['split0_train_score', 'split1_train_score',
+                'split2_train_score', 'mean_train_score', 'std_train_score']:
+        msg = (
+            'You are accessing a training score ({!r}), '
+            'which will not be available by default '
+            'any more in 0.21. If you need training scores, '
+            'please set return_train_score=True').format(key)
+        train_score = assert_warns_message(FutureWarning, msg,
```

Comment:
shouldn't we assert that there is no warning for the other ``val``s and that there is no warning for the other keys for ``'warn'``?

Otherwise LGTM.